### PR TITLE
feat: add ensure-custom-hooks-using-other-hooks

### LINKED ISSUE #1: Rule Request: `ensure-custom-hooks-using-other-hooks`
**Valid**

```ts
const useData = (key) => {
  // Valid, because useData is using other React Hooks.
  return useSWR(key);
}
```

**Invalid**

```ts
const useClassnames = (obj) => {
  // Invalid, because useClassnames doesn't use any other React Hooks.
  var k, cls='';
  for (k in obj) {
    if (obj[k]) {
      cls && (cls += ' ');
      cls += k;
    }
  }
  return cls;
}
```

### PR DESCRIPTION
**Pull request by hyoban**
fix #1 

This is my first time trying to write custom eslint rules and I'm not quite sure if the way to go about it is appropriate. If you find some tests that don't pass, you can add them and I'll try to fix them.

### CODE REVIEW

**Comment by Rel1cx** [comment]
Glad to see you're trying to implement this, but I need to be sure of the proper way to do it first, and when I'm done with debug/hooks, I'll see if we should use.

**Comment by hyoban** [comment]
I will check and learn from that PR first.

**Comment by Rel1cx** [comment]
> I will check and learn from that PR first.

Minor updates, see https://github.com/eslint-react/eslint-react/commit/8cd3f526503f1f9bf8b86a615dc74b20a5810f7c

File: packages/eslint-plugin-hooks/src/rules/ensure-custom-hooks-using-other-hooks.md
Comment (Rel1cx) [inline packages/eslint-plugin-hooks/src/rules/ensure-custom-hooks-using-other-hooks.md:1]: Add further reading at the end of the document to guide user to these two

[react.dev: invalid-hook-call-warning (the first note)](https://react.dev/warnings/invalid-hook-call-warning)
[react.dev: should-all-functions-called-during-rendering-start-with-the-use-prefix (the deep dive)](https://react.dev/learn/reusing-logic-with-custom-hooks#should-all-functions-called-during-rendering-start-with-the-use-prefix)

**Review by Rel1cx — CHANGES_REQUESTED** [review]

**Review by Rel1cx — APPROVED** [review]

**Comment by Rel1cx** [comment]
I've updated the hooks-collector implementation to support more complex cases, such as nested hooks.

**Pull request merged**